### PR TITLE
Taking in use app config

### PR DESCRIPTION
**Pull request by truhacevkir**
Reading cache setting from AppConfig instead of hardcoded boolean in environment variables.

Today:
- Cache setting has to be updated in 3 different lambdas that are using cache.
New:
- Cache setting is stored as ssm parameter. AppConfig reads this parameter and manually deployes this setting. Cache setting has to be updated single place in application. 

See documentation and give feedback:
https://sikt.atlassian.net/wiki/spaces/NVAP/pages/3485204497/Publication-channel+cache

### CODE REVIEW

File: src/test/java/no/sikt/nva/pubchannels/channelregistrycache/db/service/CacheServiceTestSetup.java
Comment (truhacevkir) [inline src/test/java/no/sikt/nva/pubchannels/channelregistrycache/db/service/CacheServiceTestSetup.java:1]: Renaming this file to include Test in class name

File: template.yaml [hunk 1]
```diff
@@ -172,6 +188,17 @@ Resources:
...
+  AppConfigActionsPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - appconfig:GetConfiguration
+            Resource: '*'
```
Comment (truhacevkir) [inline template.yaml:200]: https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsappconfig.html#awsappconfig-environment

It seems like AWS does not support resource-level permission for appconfig:GetConfiguration.

File: build.gradle [hunk 2]
```diff
@@ -10,4 +10,8 @@ test {
     environment("CHANNEL_REGISTER_CACHE_BUCKET", "someBucket")
     environment("CHANNEL_REGISTER_CACHE_S3_OBJECT", "someFile")
     environment("TABLE_NAME", "someTable")
+    environment("APPLICATION_CONFIG_NAME", "comeConfing")
+    environment("APPLICATION_CONFIG_PROFILE_NAME", "comeProfile")
```
Comment (brinxmat) [inline build.gradle:14]: "come"? I know itc not nececcary to komment thic :D спасибо, брат!

File: src/main/java/no/sikt/nva/pubchannels/utils/ApplicationConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import static nva.commons.core.attempt.Try.attempt;
+import com.fasterxml.jackson.databind.JsonNode;
+import no.unit.nva.commons.json.JsonUtils;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import software.amazon.awssdk.services.appconfig.AppConfigClient;
+import software.amazon.awssdk.services.appconfig.model.GetConfigurationRequest;
+
+public class ApplicationConfiguration {
```
Comment (brinxmat) [inline src/main/java/no/sikt/nva/pubchannels/utils/ApplicationConfiguration.java:11]: Can you make this implement an interface called ApplicationConfiguration. 

Replace the mocked version with a fake where relevant.

**Review by brinxmat — APPROVED** [review]

**Pull request merged**